src/catalog/catalog_entry/table_catalog_entry.cpp

```diff
@@ -441,7 +441,16 @@ unique_ptr<CatalogEntry> TableCatalogEntry::RemoveColumn(ClientContext &context,
 		return make_unique<TableCatalogEntry>(catalog, schema, (BoundCreateTableInfo *)bound_create_info.get(),
 		                                      storage);
 	}
-	auto new_storage = make_shared<DataTable>(context, *storage, removed_index);
+
+	// We need to skip generated columns because they are not in 'storage'
+	column_t storage_index = removed_index;
```

Comment:
The columns store their `StorageOid`, so this could be replaced with `columns[removed_index].StorageOid()`

Reply:
Yes, you are right.